### ISSUE TITLE
Build failures: StackOverflowError and OutOfMemoryError 

### ISSUE THREAD

**Issue by fbennett** [validator/validator#42]
Following a recommendation in #30 ([here](https://github.com/validator/validator/issues/30#issuecomment-75880351)), I have taken a stab at building the checker on a small VPS server with 1gb of main memory. It has a few other processes running, things are a little tight.

I'm getting StackOverflowError with this:
``` python
python ./build/build.py test
```
and OutOfMemoryError with this:
``` python
python ./build/build.py --stacksize=512
```
That looks like the machine just isn't big enough for the process. Would that be the right read, given that spec? (If so, I can apply for a larger VPS, but ... how big does it need to be?)

**Comment by fbennett** [comment]
Hold that thought. Still no luck via the build script, but a dumped vnu.jar just ran with the -Xss512 parameter. I'm out of my depth with Java, so it's all a little mysterious; but I seem to be off to the races for trials, at least.

**Comment by sideshowbarker** [comment]
OK. Well feel free to reopen this if needed. I really would like to get to the point where everybody can just just the vnu.jar and almost nobody other than man ever really needs to run the build script.

As far as the resource requirements, I have myself run into problems building on a older 32 bit system but I also have myself of 64-bit system with only 1GB of RAM and I can build and run the validator on that system just fine, straight from the sources and the vanilla build instructions/options.

So bottom line is that I think at a minimum you probably need a 64-bit system but if you have that, even just a 1GB system should be plenty.

**Comment by rmzelle** [comment]
Did you try just running the vnu.jar, Frank?

**Comment by fbennett** [comment]
Yes.

**Comment by rmzelle** [comment]
And the outcome was?

**Comment by fbennett** [comment]
Various failures.  StackOverflow. Version support failure in An Eclipse
object. Failure to find the serve let object. Out of memory error.

I was still shaking down the Java environment, which is alien turf.

The information isn't useful, that's why I closed the issue.

**Comment by fbennett** [comment]
@sideshowbarker, @rmzelle 

Just a note to say that it's working, and brilliantly so. I'll retrace a little when I get a chance. Looking back, I think the problems I had were a combination of Java version mismatches and the stacksize issue. I also had some trouble getting at the display through a lighttpd-based reverse proxy. After I've rested up a bit, I'll try to bundle up some notes, in case it's useful for documentation.

**Comment by sideshowbarker** [comment]
> Just a note to say that it's working, and brilliantly so.

Glad to hear it

> After I've rested up a bit, I'll try to bundle up some notes, in case it's useful for documentation.

That would be welcome, yeah
